### ISSUE TITLE
notify_entries not firing

### ISSUE THREAD

**Issue by barbecuewings** [Flexget/Flexget#1581]
### Expected behaviour:
On successful add of a torrent to transmission, a notification should be sent via Transmission.

### Actual behaviour:
Flexget seems to skip over that config line...

### Steps to reproduce:
- Step 1: ...

#### Config:
http://pastebin.com/1teNetcZ
  
#### Log:
```
<snip>
2016-12-20 16:41 VERBOSE  task          get_shows       REJECTED: `Designated Survivor` by seen plugin because entry with title `Designated Survivor` is already marked seen in the task get_shows at 2016-09-29 17:58
2016-12-20 16:41 DEBUG    seen          get_shows       Rejecting 'https://next-episode.net/marvels-iron-fist' 'Marvel's Iron Fist' because of seen 'Marvel's Iron Fist'
2016-12-20 16:41 VERBOSE  task          get_shows       REJECTED: `Marvel's Iron Fist` by seen plugin because entry with title `Marvel's Iron Fist` is already marked seen in the task get_shows at 2016-10-04 13:00
2016-12-20 16:41 DEBUG    seen          get_shows       Rejecting 'https://next-episode.net/game-of-thrones' 'Game of Thrones' because of seen 'Game of Thrones'
2016-12-20 16:41 VERBOSE  task          get_shows       REJECTED: `Game of Thrones` by seen plugin because entry with title `Game of Thrones` is already marked seen in the task get_shows at 2016-09-29 17:58
2016-12-20 16:41 VERBOSE  task          get_shows       REJECTED: `Avengers.Assemble.S01E15.Planet.Doom.WEBRiP.x264-QCF[rartv]` by torrent_alive plugin because had < 2 required seeds. (0)
2016-12-20 16:41 DEBUG    series        get_shows       adding episode S01E26 into series Guardians of the Galaxy
2016-12-20 16:41 DEBUG    series        get_shows       -> added <Episode(id=None,identifier=S01E26,season=1,number=26)>
2016-12-20 16:41 DEBUG    series        get_shows       adding release <SeriesParser(data=Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv],name=Guardians of the Galaxy,id=None,season=1,episode=26,quality=hdtv h264,proper=0,status=OK)> into episode
2016-12-20 16:41 DEBUG    series        get_shows       -> added <Release(id=None,quality=hdtv h264,downloaded=None,proper_count=0,title=Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv])>
2016-12-20 16:41 DEBUG    series        get_shows       start with episodes: [u'Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv]']
2016-12-20 16:41 DEBUG    series        get_shows       current episodes: [u'Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv]']
2016-12-20 16:41 DEBUG    series        get_shows       quality req: dvdscr+ 720p+
2016-12-20 16:41 VERBOSE  series        get_shows       Ignored `Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv]`. Does not meet quality requirement `dvdscr+ 720p+`.
2016-12-20 16:41 DEBUG    series        get_shows       no quality meets requirements
2016-12-20 16:41 VERBOSE  task          get_shows       ACCEPTED: `Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv]` by accept_all plugin
2016-12-20 16:41 DEBUG    urlrewriter   get_shows       Checking 1 entries
2016-12-20 16:41 DEBUG    download      get_shows       Accepting magnet url for Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv]
2016-12-20 16:41 VERBOSE  details       get_shows       Summary - Accepted: 1 (Rejected: 35 Undecided: 0 Failed: 0)
2016-12-20 16:41 DEBUG    torrent_alive get_shows       Not checking trackers for seeds, as torrent_seeds is already filled.
2016-12-20 16:41 DEBUG    transmission  get_shows       Successfully connected to transmission.
2016-12-20 16:41 INFO     transmission  get_shows       "Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv]" torrent added to transmission
2016-12-20 16:41 VERBOSE  list_add      get_shows       adding accepted entries into entry_list - next_episode_shows
2016-12-20 16:41 DEBUG    entry_list    get_shows       adding entry <Entry(title=Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv],state=accepted)> to list next_episode_shows
2016-12-20 16:41 DEBUG    series        get_shows       on_task_learn
2016-12-20 16:41 DEBUG    series        get_shows       marking 1 releases as downloaded for <Entry(title=Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv],state=accepted)>
2016-12-20 16:41 DEBUG    seen          get_shows       Learned 'DA6B0D241A9F7C9EA4344BA5FCE6F6380C4FC9A0' (field: torrent_info_hash, local: 0)
2016-12-20 16:41 DEBUG    seen          get_shows       Learned 'Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F[rartv]' (field: title, local: 0)
2016-12-20 16:41 DEBUG    seen          get_shows       Learned 'magnet:?xt=urn:btih:da6b0d241a9f7c9ea4344ba5fce6f6380c4fc9a0&dn=Guardians.of.the.Galaxy.S01E26.HDTV.x264-W4F%5Brartv%5D&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710&tr=udp%3A%2F%2F9.rarbg.to%3A2710&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce' (field: url, local: 0)
2016-12-20 16:41 DEBUG    task          get_shows       not running task_exit yet because task will rerun
2016-12-20 16:41 DEBUG    next_series_episodes get_shows       Guardians of the Galaxy S01E26 was accepted, rerunning to look for next ep.
2016-12-20 16:41 DEBUG    task          get_shows       Plugin next_series_episodes has requested task to be ran again after execution has completed. Reason: Look for next episode
2016-12-20 16:41 INFO     task          get_shows       Rerunning the task in case better resolution can be achieved.
2016-12-20 16:41 DEBUG    task          get_shows       executing get_shows
2016-12-20 16:41 DEBUG    task          get_shows       skipping task_start during rerun

</snip>

```
### Additional information:

- Flexget Version: 2.8.4
- Python Version: 2.7.11
- Installation method: pip
- OS and version: macOS 10.12.2

**Comment by liiight** [comment]
`not running task_exit yet because task will rerun`
 This is the reason... @gazpachoking is this still how this works in the new refactor?

**Comment by gazpachoking** [comment]
No, I put the notify event before exit, but I'm pretty sure exit is special cased to be the only one that only runs once after all reruns.

**Comment by cvium** [comment]
This should be fixed as 2.9.0 is out. Let us know if it still fails.
